src/zenml/artifact_stores/step_logging_utils.py

```diff
@@ -0,0 +1,102 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utilities for step logs."""
+
+import logging
+import os
+from typing import TYPE_CHECKING, Optional
+from uuid import UUID, uuid4
+
+from zenml.artifact_stores.base_artifact_store_logging_handler import (
+    ArtifactStoreLoggingHandler,
+)
+from zenml.io import fileio
+from zenml.logger import get_logger
+
+if TYPE_CHECKING:
+    from zenml.artifact_stores import BaseArtifactStore
+
+
+logger = get_logger(__name__)
+
+
+def prepare_logs_uri(
+    artifact_store: "BaseArtifactStore",
+    pipeline_run_id: UUID,
+    step_name: str,
+    log_key: Optional[str] = None,
+) -> str:
+    """Generates and prepares a URI for the log file for a step.
+
+    Args:
+        artifact_store: The artifact store on which the artifact will be stored.
+        pipeline_run_id: The id of the pipeline run.
+        step_name: Name of the step.
+        log_key: The unique identification key of the log file.
+
+    Returns:
+        The URI of the output artifact.
+    """
+    if log_key is None:
+        log_key = str(uuid4())
+
+    logs_base_uri = os.path.join(
+        artifact_store.path,
+        str(pipeline_run_id),
+        step_name,
+    )
+
+    # Create the dir
+    if not fileio.exists(logs_base_uri):
+        fileio.makedirs(logs_base_uri)
+
+    # Check if the file already exists, if so, raise an error
+    logs_uri = os.path.join(logs_base_uri, f"{log_key}.log")
+    if fileio.exists(logs_uri):
+        raise RuntimeError(f"Logs file {logs_uri} already exists!")
```

Comment:
Can't immediately see this, but I hope wherever we call this we catch it. I don't think a pipeline run should fail in this case. WDYT?

Reply:
Im not sure - this seems like a catastrophic failure  case to me where somebody is messing around with the logs... How else do we handle it: Just delete the old file and start a new one?

Reply:
I would say maybe we can write a warning to the preexisting file and keep going with the pipeline run?

Reply:
Now I did it in a way that if the file exists, we just remove the old one and therefore start anew